src/lv_misc/lv_font_fmt/lv_font_fmt_txt.h

```diff
@@ -0,0 +1,193 @@
+/**
+ * @file lv_font.h
+ *
+ */
+
+#ifndef LV_FONT_H
+#define LV_FONT_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*********************
+ *      INCLUDES
+ *********************/
+#ifdef LV_CONF_INCLUDE_SIMPLE
+#include "lv_conf.h"
+#else
+#include "../../../lv_conf.h"
+#endif
+
+#include <stdint.h>
+#include <stddef.h>
+#include <stdbool.h>
+
+#include "../lv_font.h"
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+/*Describe a glyph*/
+typedef struct
+{
+    uint32_t bitmap_index : 20;     /* Start index of the bitmap. A font can be max 1 MB. */
+    uint32_t adv_w :12;             /*The glyph needs this space. Draw the next glyph after this width. 8.4 format. */
+
+    uint8_t box_w;                  /*Width of the glyph's bounding box*/
+    uint8_t box_h;                  /*Height of the glyph's bounding box*/
+    int8_t ofs_x;                   /*x offset of the bounding box*/
+    uint8_t ofs_y;                  /*y offset of the bounding box. Measured from the top of the line*/
+}lv_font_glyph_dsc_fmt_txt_t;
+
+/* Map codepoints to a `glyph_dsc`s
+ * Several formats are supported to optimize memory usage
+ * See https://github.com/littlevgl/lv_font_conv/blob/master/doc/font_spec.md
+ */
+typedef struct {
+    /* First Unicode character for this range */
+    uint32_t range_start;
+
+    /* Number of Unicode characters related to this range.
+     * Last Unicode character = range_start + range_length - 1*/
+    uint16_t range_length;
+
+    /* First glyph ID (array index of `glyph_dsc`) for this range */
+    uint16_t glyph_id_start;
+
+    /*
+    According the specification there are 4 formats:
+        https://github.com/littlevgl/lv_font_conv/blob/master/doc/font_spec.md
+
+    For simplicity introduce "relative code point":
+        rcp = codepoint - range_start
+
+    and a search function:
+        search a "value" in an "array" and returns the index of "value".
+
+    Format 0 tiny
+        unicode_list == NULL && glyph_id_list == NULL
+        glyph_id = glyph_id_start + rcp
+
+    Format 0 full
+        unicode_list == NULL && glyph_id_list != NULL
+        glyph_id = glyph_id_start + glyph_id_list[rcp]
+
+    Sparse tiny
+        unicode_list != NULL && glyph_id_list == NULL
+        glyph_id = glyph_id_start + search(unicode_list, rcp)
+
+    Sparse full
+        unicode_list != NULL && glyph_id_list != NULL
+        glyph_id = glyph_id_start + glyph_id_list[search(unicode_list, rcp)]
+    */
+
+    uint16_t * unicode_list;
+
+    /* if unicode_list == NULL uint8_t *
+     * else uint16_t *
+     */
+    const void * glyph_id_list;
```

Comment:
may be `union { uint8_t u8[], uint16_t u16[]; } glyph_id_list`?

Casting from `void` in runtime looks not beautiful IMO.